lib/utils/config.ts

```diff
@@ -165,3 +176,48 @@ export async function validateDevOptionAndWarn(
 		and exposes network ports such as 2375 that allows unencrypted access to balenaEngine.
 		Therefore, development mode should only be used in private, trusted local networks.`);
 }
+
+/**
+ * Chech whether the `--secureBoot` option of commands related to OS configuration
+ * such as `os configure` and `config generate` is compatible with a given
+ * device type and balenaOS version, and print a warning regarding the
+ * consequences of using that option.
+ */
+export async function validateSecureBootOptionAndWarn(
+	secureBoot?: boolean,
+	deviceManifest?: any,
+	version?: string,
+	logger?: import('./logger'),
+) {
+	if (!secureBoot) {
+		return;
+	}
+
+	const deviceManifestObj: { [index: string]: any } = deviceManifest;
+	const deviceType = deviceManifestObj['yocto']['machine'];
+
+	const { ExpectedError } = await import('../errors');
+	const supportedDeviceTypes = ['generic-amd64'];
+	if (!supportedDeviceTypes.includes(`${deviceType}`)) {
+		throw new ExpectedError(
+			`Error: The '--secureBoot' option is not supported for the '${deviceType}' device Type. Supported device types include ${supportedDeviceTypes.join(
+				', ',
+			)}.`,
+		);
+	}
+
+	const supportedFromVersion = '2.114.21';
+	if (!semver.satisfies(version, `>=${supportedFromVersion}`)) {
```

Comment:
This would give false positives in case the OS version is an ESR.
For that case we need
* to use `sdk.models.os.getAllOsVersions(deviceType)` to fetch the OS versions for the DT
* find the OS version that was provided
* use the `basedOnVersion` for this comparison.

Reply:
thanks for raising this @thgreasi . We do not release4 ESR releases at this moment for the only DT that supports this (generic-amd64), but I agree the solution needs to cater for ESR too.

I was wondering whether we could make use of the `meta-balena-base` label to compare when dealing with ESR, that is, we know the feature was introduced in `2.114.21` so any ESR whose `meta-balena-base` is above that will support it.

Reply:
@alexgg are you referring to the `meta-balena-base` release tag? B/c that's exactly what the `basedOnVersion` holds in my above suggestion (as returned by `sdk.models.os.getAllOsVersions(deviceType)`).

If this is the only blocker, we can just for now switch `>=` to `^` in the `satisfies` call and generalize this post-MVP

Reply:
hi @thgreasi I pushed an implementation of the above. It's not completely tested as I  want to try an use the OS contract instead as I mentioned above. Will come back to this if I don't manage to make the contract work in reasonable time.